### PR TITLE
Feature request - Handling Elastic Search term suggestions in the Search Response

### PR DESCRIPTION
**Pull request by jpetar**
### Short description and why it's useful

This change would allow us to implement the term suggestions from Elastic Search by handling _suggest_ in the response.

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

File: core/lib/search/adapter/api/searchAdapter.ts [hunk 1]
```diff
@@ -74,7 +76,7 @@ export class SearchAdapter {
       httpQuery.request = JSON.stringify(ElasticsearchQueryBody)
     }
     url = url + '/' + encodeURIComponent(Request.index) + '/' + encodeURIComponent(Request.type) + '/_search'
-    url = url + '?' + queryString.stringify(httpQuery)
+    url = url + '?' + buildURLQuery(httpQuery)
```
Comment (pkarw) [inline core/lib/search/adapter/api/searchAdapter.ts]: Why not queryString.steingify?

**Comment by pkarw** [comment]
Please provide us with some screens of: before / after. Please also make sure you’ve applied the changes to the most recent develop branch as it seems like the changes of “hasOwnProperty” instead of just value checking and replacement of queryString.stringify with buildUrlQuwry are kind of reverting the most recent changes (from 1.7, 1.8). Did you apply your feature changes to vs 1.8.5 or some earlier version?

File: core/lib/search/adapter/api/searchAdapter.ts [hunk 2]
```diff
@@ -92,12 +94,13 @@ export class SearchAdapter {
     if (resp.hasOwnProperty('hits')) {
       return {
         items: map(resp.hits.hits, hit => {
-          return Object.assign(hit._source, { _score: hit._score, slug: hit._source.slug ? hit._source.slug : ((hit._source.hasOwnProperty('url_key') && rootStore.state.config.products.useMagentoUrlKeys) ? hit._source.url_key : (hit._source.hasOwnProperty('name') ? slugify(hit._source.name) + '-' + hit._source.id : '')) }) // TODO: assign slugs server side
+          return Object.assign(hit._source, { _score: hit._score, slug: (hit._source.hasOwnProperty('url_key') && rootStore.state.config.products.useMagentoUrlKeys) ? hit._source.url_key : (hit._source.hasOwnProperty('name') ? slugify(hit._source.name) + '-' + hit._source.id : '') }) // TODO: assign slugs server side
```
Comment (pkarw) [inline core/lib/search/adapter/api/searchAdapter.ts]: Why this change is required?

File: core/lib/search/adapter/graphql/processor/processType.ts [hunk 3]
```diff
@@ -8,15 +8,16 @@ export function processESResponseType (resp, start, size): SearchResponse {
     items: map(resp.hits.hits, hit => {
       return Object.assign(hit._source, {
         _score: hit._score,
-        slug: hit._source.slug ? hit._source.slug : ((hit._source.hasOwnProperty('url_key') && rootStore.state.config.products.useMagentoUrlKeys)
```
Comment (pkarw) [inline core/lib/search/adapter/graphql/processor/processType.ts]: Why this change is required?

File: core/lib/search/adapter/graphql/processor/processType.ts [hunk 4]
```diff
@@ -8,15 +8,16 @@ export function processESResponseType (resp, start, size): SearchResponse {
...
+        slug: (hit._source.hasOwnProperty('url_key') && rootStore.state.config.products.useMagentoUrlKeys)
           ? hit._source.url_key
-          : (hit._source.hasOwnProperty('name') ? slugify(hit._source.name) + '-' + hit._source.id : ''))
```
Comment (pkarw) [inline core/lib/search/adapter/graphql/processor/processType.ts]: Why this change is required?

File: core/lib/search/adapter/graphql/processor/processType.ts [hunk 5]
```diff
@@ -30,17 +31,18 @@ export function processProductsType (resp, start, size): SearchResponse {
         options['_score'] = item._score
         delete item._score
       }
-      options['slug'] = item.slug ? item.slug : ((item.url_key &&
+      options['slug'] = (item.hasOwnProperty('url_key') &&
       rootStore.state.config.products.useMagentoUrlKeys)
-        ? item.url_key : (item.name
-          ? slugify(item.name) + '-' + item.id : ''))
+        ? item.url_key : (item.hasOwnProperty('name')
```
Comment (pkarw) [inline core/lib/search/adapter/graphql/processor/processType.ts]: Why this change is required?

File: core/lib/search/adapter/graphql/processor/processType.ts [hunk 6]
```diff
@@ -30,17 +31,18 @@ export function processProductsType (resp, start, size): SearchResponse {
         options['_score'] = item._score
         delete item._score
       }
-      options['slug'] = item.slug ? item.slug : ((item.url_key &&
```
Comment (pkarw) [inline core/lib/search/adapter/graphql/processor/processType.ts]: Why this change is required?

**Comment by jpetar** [comment]
I've closed this pull request because it includes unwanted changes.
